### PR TITLE
Create helm chart for weblate

### PR DESCRIPTION
**Pull request by mcfedr**
I've made an am using this helm chart for weblate, I thought it might be of interest, so I'm sharing it.

Its probably not perfect for the general use case, for that would probably be good to add the option of creating a postgres deployment, but I am thinking someone with such an interest might add it.

### CODE REVIEW

**Comment by nijel** [comment]
Thanks! Can you please briefly describe using it into our deployment docs? You can find it here: https://github.com/WeblateOrg/weblate/blob/master/docs/admin/deployments.rst

**Comment by mareksuscak** [comment]
This looks great but I would recommend using a `stable/redis` chart to deploy redis. There's no point in replicating what it already does, is there? That said, unless weblate requires specific configuration which can anyway be achieved with the official chart.

File: charts/weblate/templates/statefulset.yaml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+      tolerations:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+      securityContext:
+        fsGroup: 1000
+      volumes:
+        - name: data
+          {{- if .Values.persistence.enabled }}
+          persistentVolumeClaim:
+            claimName: {{ include "weblate.fullname" . }}
```
Comment (mareksuscak) [inline charts/weblate/templates/statefulset.yaml]: I know weblate probably doesn't support horizontal scaling at the moment but for StatefulSets it's more natural to [use `volumeClaimTemplates`](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#components). What do you think?

**Comment by rdxmb** [comment]
@mareksuscak 

> We could also add an optional stable/postgresql dependency if anyone wants to host it all on k8s even though I would discourage everyone from doing so.

We are doing so. What's the problem?

**Comment by mareksuscak** [comment]
@rdxmb We're not. I couldn't find it. Can you point me to the requirements file?

**Comment by rdxmb** [comment]
sorry, my comments above from last week were not submitted. So the comments are actually from "now".

**Comment by rdxmb** [comment]
@mareksuscak requirements to what?

File: charts/weblate/values.yaml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  repository: weblate/weblate
+  tag: latest
+  pullPolicy: Always
+
+nameOverride: ""
+fullnameOverride: ""
+
+service:
+  type: ClusterIP
+  loadBalancerIP: ~
```
[the remark this answers is not in the document]
Comment (mareksuscak) [reply charts/weblate/values.yaml]: It's a null character. Denotes an empty value.

[the remark this answers is not in the document]
Comment (rdxmb) [reply charts/weblate/values.yaml]: So it is equal to
`loadBalancerIP: "" `
?

**Comment by mareksuscak** [comment]
@rdxmb https://helm.sh/docs/developing_charts/#chart-dependencies

File: charts/weblate/values.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (mareksuscak) [reply charts/weblate/values.yaml]: Yup I think so.

**Comment by rdxmb** [comment]
@mareksuscak oh, there is none. But I use weblate with my own helm package and also redis and postgres in kubernetes. That's what you wrote

> I would discourage everyone from doing so.

Why?

File: charts/weblate/values.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (mcfedr) [reply charts/weblate/values.yaml]: No, its a null not an empty object.

**Comment by mcfedr** [comment]
@mareksuscak

> This looks great but I would recommend using a `stable/redis` chart to deploy redis.

Probably makes sense, basically I don't really like that it uses `bitnami/redis` instead of just `redis`, but really doesnt matter much.

> We could also add an optional stable/postgresql dependency

I agree, I'm sure this will be useful to someone, personally I have no interest in doing so, and testing with it - so I'd rather consider this a PR without it, and let someone else add it in after merging.

File: charts/weblate/templates/statefulset.yaml (hunk 1 above)
Comment (mcfedr) [reply charts/weblate/templates/statefulset.yaml]: I've never used `volumeClaimTemplates` - so ill have to have a look.

File: charts/weblate/templates/ingress.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
+{{- if .Values.ingress.enabled -}}
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: {{ include "weblate.fullname" . }}
+  labels:
+{{ include "weblate.labels" . | indent 4 }}
+  annotations:
```
[the remark this answers is not in the document]
Comment (mcfedr) [reply charts/weblate/templates/ingress.yaml]: I guess so .- I've just pushed my, quite personallised, template, so i guess I could simplify this.

File: charts/weblate/templates/statefulset.yaml [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          env:
+            - name: WEBLATE_ALLOWED_HOSTS
+              value: {{ .Values.host }},*
+            - name: WEBLATE_DEBUG
+              value: "0"
+            - name: WEBLATE_LOGLEVEL
+              value: "INFO"
+            - name: POSTGRES_HOST
```
[the remark this answers is not in the document]
Comment (mcfedr) [reply charts/weblate/templates/statefulset.yaml]: The chart doesnt include postgres, see https://github.com/WeblateOrg/docker-compose/pull/8#issuecomment-542058346

**Comment by rdxmb** [comment]
I also would prefer the official redis image. Probably this could easy be changed.

My suggestion for now is that the README.md should include, that this chart does not include redis and postgres by now. So we could get this PR merged.

File: charts/weblate/templates/pvc.yaml [hunk 5]
```diff
@@ -0,0 +1,21 @@
+{{- if .Values.persistence.enabled }}
+apiVersion: v1
+kind: PersistentVolumeClaim
```
[the remark this answers is not in the document]
Comment (mareksuscak) [reply charts/weblate/templates/pvc.yaml]: You can get some inspiration on how to do that [here](https://github.com/helm/charts/blob/master/stable/redis/templates/redis-master-statefulset.yaml#L376-L394).

**Comment by Antiarchitect** [comment]
Is there any progress in merging this?

**Comment by nijel** [comment]
Given my zero Helm knowledge, I will merge this once there will to be consensus it's correct (what doesn't seem to be right now).

**Comment by MPV** [comment]
For anyone else ending up here, see more recent discussions (and work) here:
https://github.com/WeblateOrg/weblate/issues/1205 
https://github.com/WeblateOrg/weblate/issues/1205#issuecomment-623008218

**Pull request closed without merging**